### ISSUE TITLE
Push registration token refresh forces default ClientOptions

### ISSUE THREAD

**Issue by tcard** [ably/ably-java#562]
[Originally](https://github.com/ably/ably-java/blob/b9011d583782a3f6fb893622be63989f6965ef60/android/src/main/java/io/ably/lib/fcm/AblyFirebaseInstanceIdService.java), you had to pass a `AblyRest` instance when refreshing registration tokens, but now, the `AblyRest` is [instanced by us, using the `deviceIdentityToken` as key, with default options](https://github.com/ably/ably-java/blob/8e26a834d9de39b3d4c9b822373dce551427e763/android/src/main/java/io/ably/lib/push/ActivationContext.java#L66-L67). So, if the client relies on non-default options (custom fallbacks, custom push registration), token refresh won't work.

**Comment by tcard** [comment]
@paddybyers WDYT?

**Comment by paddybyers** [comment]
I agree this is a problem - I thought I'd already raised an issue for it somewhere.
